plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeFileScanTest.kt

```diff
@@ -166,6 +211,74 @@ class CodeWhispererCodeFileScanTest : CodeWhispererCodeScanTestBase(PythonCodeIn
         }
     }
 
+    @Test
+    fun `test run() - measure CPU and memory usage`() {
+        // Set up CPU and Memory monitoring
+        val runtime = Runtime.getRuntime()
+        val bean = ManagementFactory.getThreadMXBean()
+        val startCpuTime = bean.getCurrentThreadCpuTime()
+        val startMemoryUsage = runtime.totalMemory() - runtime.freeMemory()
+        val startSystemTime = System.nanoTime()
+
+        // Run the code scan
+        runBlocking {
+            codeScanSessionSpy2.run()
+        }
+
+        // Calculate CPU and memory usage
+        val endCpuTime = bean.getCurrentThreadCpuTime()
+        val endMemoryUsage = runtime.totalMemory() - runtime.freeMemory()
+        val endSystemTime = System.nanoTime()
+
+        val cpuTimeUsedNanos = endCpuTime - startCpuTime
+        val cpuTimeUsedSeconds = cpuTimeUsedNanos / 1_000_000_000.0
+        val elapsedTimeSeconds = (endSystemTime - startSystemTime) / 1_000_000_000.0
+
+        val memoryUsed = endMemoryUsage - startMemoryUsage
+        val memoryUsedInMB = memoryUsed / (1024.0 * 1024.0) // Converting into MB
+
+        // Calculate CPU usage in percentage
+        val cpuUsagePercentage = (cpuTimeUsedSeconds / elapsedTimeSeconds) * 100
+
+        assertThat(cpuTimeUsedSeconds).isLessThan(5.0)
+        assertThat(cpuUsagePercentage).isLessThan(30.0)
+        assertThat(memoryUsedInMB).isLessThan(200.0) // Memory used should be less than 200MB
+    }
+
+    @Test
+    fun `test run() - measure CPU and memory usage with payload of 150KB`() {
+        // Set up CPU and Memory monitoring
+        val runtime = Runtime.getRuntime()
+        val bean = ManagementFactory.getThreadMXBean()
+        val startCpuTime = bean.getCurrentThreadCpuTime()
+        val startMemoryUsage = runtime.totalMemory() - runtime.freeMemory()
+        val startSystemTime = System.nanoTime()
+
+        // Run the code scan
+        runBlocking {
+            codeScanSessionSpy3.run()
+        }
+
+        // Calculate CPU and memory usage
+        val endCpuTime = bean.getCurrentThreadCpuTime()
+        val endMemoryUsage = runtime.totalMemory() - runtime.freeMemory()
+        val endSystemTime = System.nanoTime()
+
+        val cpuTimeUsedNanos = endCpuTime - startCpuTime
+        val cpuTimeUsedSeconds = cpuTimeUsedNanos / 1_000_000_000.0
+        val elapsedTimeSeconds = (endSystemTime - startSystemTime) / 1_000_000_000.0
```

Comment:
why are we bothering using a high-resolution time source if we're just going to convert it back to seconds?